packages/graphql_server/src/dataAggregation.ts

```diff
@@ -0,0 +1,126 @@
+  const repoGHdata: GitHubInfo | null = await getRepoInfo(
+    query,
+    'Bearer ' + process.env.GITHUB_API_TOKEN
+  )
+
+  if (repoGHdata === null) {
+    console.error('Could not get GitHub data for repo', name)
+    return null
+  }
+
+  const organizationID = await getOrganizationID(owner)
+
+  // @Todo aggregate more data for the repo
+
+  repoInfo = {
+    name: repoGHdata.name,
+    about: repoGHdata.description,
+    star_count: repoGHdata.stargazerCount,
+    issue_count: repoGHdata.issues.totalCount,
+    fork_count: repoGHdata.forkCount,
+    pull_request_count: repoGHdata.pullRequests.totalCount,
+    contributor_count: 1,
+    github_url: repoGHdata.url,
+    website_url: repoGHdata.homepageUrl,
+    owned_by: organizationID ?? '634b6eb5-30c8-4818-81d3-e1d98cb0b2c7',
+    is_bookmarked: false
+  }
+
+  return repoInfo
+}
+
+/**
+ * Returns the organization id for the given organization name.
+ * It tries to get that id from the DB. If it is not in the DB it gets the data from github and inserts it into the DB.
+ * @param {string} owner - The name of the organization.
+ * @returns {string} The id of the organization.
+ */
+const getOrganizationID = async (owner: string) => {
+  const { data: organization, error: getOrganizationError } = await supabase
+    .from('organization')
+    .select('id')
+    .eq('name', owner)
+  getOrganizationError &&
+    console.error('Error getting organization', owner, 'from database: \n', getOrganizationError)
+
+  // if a organization with this name is already in the database return the id
+  if (organization && organization[0]) {
+    return organization[0].id
+  }
+
+  // if not get the data from github and insert it into the database
+  const query = `query {
+      organization(login: "${owner}") {
+        login
+        name
+        avatarUrl
+        repositories {totalCount}
+        email
+        websiteUrl
+        twitterUsername
+        url
+    }
+    }`
+
+  const organizationGHData: GitHubOrganization | null = await getOrganizationInfo(
+    query,
+    'Bearer ' + process.env.GITHUB_API_TOKEN
+  )
+
+  if (organizationGHData !== null) {
+    const organizationDataDBFormat: OrganizationInsertion = {
+      name: organizationGHData.name,
+      login: organizationGHData.login,
+      avatar_url: organizationGHData.avatarUrl,
+      repository_count: organizationGHData.repositories.totalCount,
+      email: organizationGHData.email,
+      website_url: organizationGHData.websiteUrl,
+      twitter_username: organizationGHData.twitterUsername,
+      github_url: organizationGHData.url
+    }
+
+    const { error: insertOrganizationError } = await supabase
+      .from('organization')
+      .insert([organizationDataDBFormat])
+    insertOrganizationError &&
+      console.error('Error inserting organization into database: \n', insertOrganizationError)
+
+    const { data: orga } = await supabase.from('organization').select('id').eq('name', owner)
+
+    if (orga && orga[0]) {
```

Comment:
if (orga?.[0]?.id) return orga?.[0]?.id; what is being returned in every other case? Shouldn't this throw an error then?

Reply:
Good point, I decided to return null in this case. Actually this case should never happen. Because if the organization does not exist it will be inserted.